### PR TITLE
Document CUDA library dependencies for python wheel

### LINKED ISSUE #700: Document installation of the necessary CUDA libraries to use GPU backends in Python wheels
While cuQuantum is defined as a wheel dependency, cuQuantum depends on CUDA libraries that are not automatically installed when installing the wheel. The documentation for installing CUDA Quantum from PyPI needs to be updated to clarify what CUDA libraries need to be installed separately. Additionally, the Readme that is included with the Python wheels needs to be updated to point to that documentation.

### PR DESCRIPTION
**Pull request by anthony-santana**
### Description

Closes #700

### CODE REVIEW

File: docs/sphinx/using/simulators.rst [hunk 1]
```diff
@@ -90,3 +90,70 @@ options to :code:`nvq++`
...
+our GPU enabled simulation backends.
+
+System Level Dependencies
+++++++++++++++++++++++++++
+
+The system level dependencies for these backends include:
+* An NVIDIA GPU with compute capability 7.0+
+* Driver: Linux (450.80.02+ for CUDA 11, 525.60.13+ for CUDA 12)
+* CUDA Toolkit 11.x or newer
+* Python 3.9+
```
Comment (schweitzpgi) [inline docs/sphinx/using/simulators.rst:107]: Should this be a link to the tables in https://github.com/NVIDIA/cuda-quantum/pull/707 ? Should the additional information here be added to those tables?

File: docs/sphinx/using/simulators.rst [hunk 2]
```diff
@@ -90,3 +90,70 @@ options to :code:`nvq++`
...
+
+Note: After installing the CUDA Toolkit, you must follow these `post-installation instructions
+<https://docs.nvidia.com/cuda/archive/11.8.0/cuda-installation-guide-linux/index.html#post-installation-actions>`__.
+
+Alternatively, the programmer may install the following Python packages:
+* nvidia-nvtx-vcu11 
+* nvidia-cuda-runtime-cu11
+
+You must then export the location of the .so files for these packages to your :code:`LD_LIBRARY_PATH`.
+To find the local installation path of these packages, you may run
```
Comment (schweitzpgi) [inline docs/sphinx/using/simulators.rst:129]: nit: consider rephrasing to eliminate use of pronouns like you and we.

**Comment by schweitzpgi** [comment]
LGTM. Directions seem easy to follow, etc.

File: docs/sphinx/using/simulators.rst [hunk 3]
```diff
@@ -90,3 +90,70 @@ options to :code:`nvq++`
...
+
+    apt-get install cuda-11-8 # Ubuntu
+    # -- or -- 
+    dnf install cuda-11-8.x86_64 # RHEL 9
+
+Note: After installing the CUDA Toolkit, you must follow these `post-installation instructions
+<https://docs.nvidia.com/cuda/archive/11.8.0/cuda-installation-guide-linux/index.html#post-installation-actions>`__.
+
+Alternatively, the programmer may install the following Python packages:
+* nvidia-nvtx-vcu11
```
Comment (bettinaheim) [inline docs/sphinx/using/simulators.rst:125]:
```suggestion
* nvidia-nvtx-cu11 
```

File: docs/sphinx/install.rst [hunk 4]
```diff
@@ -132,6 +132,11 @@ for all releases `here <https://github.com/NVIDIA/cuda-quantum/releases>`__.
 For more information about building a CUDA Quantum Python wheel from source, see the 
 `README <https://github.com/NVIDIA/cuda-quantum/blob/main/python/README.md>`__.
 
+The CUDA Quantum Python wheels, by default, do not come installed with our GPU-enabled
```
Comment (bettinaheim) [inline docs/sphinx/install.rst:135]: The Python wheels do come with all backends now, but they require an existing CUDA installation.

File: docs/sphinx/using/simulators.rst [hunk 5]
```diff
@@ -90,3 +90,70 @@ options to :code:`nvq++`
...
+The system level dependencies for these backends include:
+* An NVIDIA GPU with compute capability 7.0+
+* Driver: Linux (450.80.02+ for CUDA 11, 525.60.13+ for CUDA 12)
+* CUDA Toolkit 11.x or newer
+* Python 3.9+
+
+For more information on the installation of NVIDIA Driver's for Ubuntu, see
+`here <https://help.ubuntu.com/community/NvidiaDriversInstallation>`__.
+
+For installation of the CUDA toolkit, programmers have two options. The first
```
Comment (bettinaheim) [inline docs/sphinx/using/simulators.rst:112]: I would move this to the install.rst starting from this line (the lines before that are covered by Eric's PR that states the compatibility). It's probably easiest if I create a PR to show what I mean.

File: docs/sphinx/using/simulators.rst (hunk 1 above)
Comment (anthony-santana) [reply docs/sphinx/using/simulators.rst]: I've removed this and am linking to the table now. I've also added the python version to the "supported systems" section

File: docs/sphinx/install.rst [hunk 6]
```diff
@@ -167,6 +170,8 @@ The following table summarizes the required components.
       - Linux
     * - Tested Distributions
       - CentOS 8; Debian 11, 12; Fedora 38; OpenSUSE/SELD/SLES 15.5; RHEL 8, 9; Rocky 8, 9; Ubuntu 22.04
+    * - Python versions
+      - 3.9+
```
Comment (bettinaheim) [inline docs/sphinx/install.rst:174]:
```suggestion
      - 3.8+
```

**Review by bettinaheim — APPROVED** [review]

**Pull request merged**